cronjobs/add_authority_external_ref._automation.py

```diff
@@ -0,0 +1,183 @@
+
+"""
+
+import os
+import re
+import requests
+import util
+from urllib.request import urlopen
+import traceback
+from tqdm import tqdm
+import functools
+import shutil
+
+# this script needs:
+# 1. requests
+# 2. tqdm
+
+# Start global variable
+
+url = "https://data.dnb.de/opendata/"
+config_file = "/mnt/ZE020150/FID-Entwicklung/ub_tools/config_file_add_authority_ext_ref.cnf"
+share_folder = "/mnt/ZE020150/FID-Entwicklung/ub_tools/"
+# config_file = "/tmp/config_file_add_authority_ext_ref.cnf"
+# share_folder = "/tmp/"
+newer_file_name = ""
+gnd_wiki_file = "gnd_to_wiki.csv"
+current_file_date_integer = 0
+last_file_update_date_int = 0
+input_file_name_for_add_auth = "input_file_for_add_authority_external_ref.txt"
+
+
+# End global variable
+
+
+def DownloadTheFile(file_name):
+    url_path = f'{url}{file_name}'
+    target_file = f'/tmp/{file_name}'
+    dw = requests.get(url_path, stream=True)
+    if dw.status_code != 200:
+        dw.raise_for_status()  # Will only raise for 4xx codes, so...
+        raise RuntimeError(
+            f"Request to {url_path} returned status code {dw.status_code}")
+
+    file_size = int(dw.headers.get('Content-Length', 0))
+
+    desc = "(Unknown total file size)" if file_size == 0 else ""
+    dw.raw.read = functools.partial(
+        dw.raw.read, decode_content=True)  # Decompress if needed
+    with tqdm.wrapattr(dw.raw, "read", total=file_size, desc=desc) as r_raw:
+        f = open(target_file, "wb")
+        shutil.copyfileobj(r_raw, f)
+    return True
+
+
+def UpdateConfigFile(config_file, successful_generate_date):
+    add_auth_ext_ref_config_file = open(config_file, "w+")
+    add_auth_ext_ref_config_file.writelines(
+        [str(successful_generate_date), "\n"])
+
+
+def IsThereANewlyReleasedFile():
+    file_name = ""
+    global config_file
+    global current_file_date_integer
+    global last_file_update_date_int
+    global newer_file_name
+
+    base_name = "authorities-gnd-person_lds_"
+    webpage_object = urlopen(url)
+    html_bytes = webpage_object.read()
+    html = html_bytes.decode("utf-8")
+
+    if re.search('authorities-gnd-person_lds_.*\.jsonld\.gz', html):
+        file_name = re.search(
+            'authorities-gnd-person_lds_.*\.jsonld\.gz', html).group(0)
+        current_file_date_integer = int(
+            file_name[len(base_name):len(base_name) + 8])
+        if (os.path.exists(config_file)):
+            add_auth_ext_ref_config_file = open(config_file, "r")
+            read_line_1 = add_auth_ext_ref_config_file.readline()
+            add_auth_ext_ref_config_file.close
+
+            if (read_line_1.strip()):
+                last_file_update_date_int = int(read_line_1)
+
+            if (last_file_update_date_int < current_file_date_integer):
+                print(f"Last update date: {last_file_update_date_int}")
+                print(
+                    f"current version on the web: {current_file_date_integer}")
+                return True
```

Comment:
Can we return the current_file_date_integer instead of just returning True/False? This way we could also get rid of one more global variable.

Reply:
I will change the logic and may be difficult to read

Reply:
I don't think so, but we can discuss it verbally if you like.